workspaces/util/src/__tests__/chunkBySize-test.ts

```diff
@@ -0,0 +1,42 @@
+import { describe, expect, it } from "@jest/globals";
+import { chunkBySize } from "../chunkBySize";
+
+describe("chunkBySize", () => {
+  it("splits an input array into smaller arrays", () => {
+    const result = chunkBySize([1, 2, 3, 4], 2);
+    expect([...result]).toStrictEqual([
+      [1, 2],
+      [3, 4],
+    ]);
+  });
+
+  it("handles an array not perfectly divisible by chunk size", () => {
+    const result = chunkBySize([1, 2, 3], 2);
+    expect([...result]).toStrictEqual([[1, 2], [3]]);
+  });
+
+  it("throws a TypeError if array contains undefined elements (sparse array)", () => {
+    expect(() => {
+      const result = chunkBySize([1, , 3], 2); // Sparse array
```

Comment:
We have a lint rule against sparse arrays in this codebase, so you will probably need to add an ESLint disable to test this:

```suggestion
      // eslint-disable-next-line no-sparse-arrays
      const result = chunkBySize([1, , 3], 2); // Sparse array
```